### DISCUSSION TITLE
[注意] 📢GitHub Issue 中的图片链接开始有期限了，应该停止在markdown直接引用？

### DISCUSSION THREAD

**Discussion by tyn1998** [X-lab2017/open-wonderland#353] [1️⃣ 新鲜事]
可能已经有好几个礼拜或好几个月了，那会儿我已经发现长时间打开的Issue页面，右击”在新标签页打开图片“会得到这个结果：

![image](https://github.com/X-lab2017/open-wonderland/assets/32434520/b2955819-4da2-4203-9d26-fba104097993)

刷新issue页面后，又能正常访问图片了。说明同一张图片的链接是动态刷新的，每次刷新后的链接都有期限。这种情况下，如果在markdown直接引用具有时效性的链接，就存在风险了。@X-lab2017/x-lab 

下面这种原始链接还是靠谱的：

![image](https://github.com/X-lab2017/open-wonderland/assets/32434520/fdd500ca-fcf6-4943-a891-a4a55e8e6e09)

浅浅搜索了下，除了[这个](https://www.reddit.com/r/github/comments/13foas2/comment/jjw56y7/?utm_source=share&utm_medium=web2x&context=3)reddit评论，没大找到相关话题。泰炜有耳闻吗 @TieWay59

**Comment by frank-zsy** [comment]
最近也发现有这种情况，看来可能是太多人把这个直接当图床用了，年报里可能就有些图是直接引用了这种链接的。

**Comment by TieWay59** [comment]
https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/attaching-files

https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/about-anonymized-urls?platform=windows

https://github.blog/changelog/2023-05-09-more-secure-private-attachments/

省流： 

- 图片渲染链接都是 gh 后台 camo 经过混淆的隐私链接，用于保护真 url 暴露出的隐私信息。
- 隐私链接有多个类型（以不同子域区分），其中你遇到问题的是 `private-user-images` 。
- `private-user-images` 顾名思义，此类隐私链与你的个人用户登录令牌绑定，只有你登录过 gh 的浏览器才能在外部浏览，否则只能限制在 github 内部浏览。
- 除了能修改你文本的人，其他人无法获取你图片的真链接，也就实现了保护。
- 一个可能的暗改是现在默认 issue 评论图片都会被处理成 `private-user-images` 的链接（以前可能不是的）。
- 如果观察真链接的图片地址，实际上资源在 `github-production-user-asset-6210df.s3.amazonaws.com` （居然是亚马逊云的服务，微软爸爸去哪儿？）。gh 他们内部对假链接只做短期的缓存，一段时间没访问直接丢弃 404 也非常合理。

以下图片我就是通过打开你文本的编辑界面直接获取真链接的效果：

![](https://github.com/X-lab2017/open-wonderland/assets/32434520/b2955819-4da2-4203-9d26-fba104097993)

左右分别是我打开我的隐私链接和你的隐私链接的结果：

![image](https://github.com/X-lab2017/open-wonderland/assets/40034603/f383f060-84c9-4c7b-a9f4-070a015e3d9a)

其他：

我觉得就算不知道，也没理由依赖 gh 的图床。最好的办法还是图片都放在一个仓库里保存。这个问题在我搞博客的时候考虑过，但没深究。 @tyn1998

**Comment by tyn1998** [comment]
@TieWay59 入木三分，感谢！！仓库对我来说还是略重了点，下次评论区有什么我想要引用的但是我没有edit权限的图片，我就复制粘贴到comment框里得到一个新的链接好了:D
